### PR TITLE
Add `queryKey` to Script Library

### PR DESCRIPTION
**Pull request by bame-da**
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-script/daml/Daml/Script.daml [hunk 1]
```diff
@@ -125,6 +126,16 @@ queryContractId p c = lift $ Free $ QueryContractId QueryContractIdPayload with
   tplId = templateTypeRep @t
   cid = coerceContractId c
   continue = pure . fmap (fromSome . fromAnyTemplate)
+  
+-- | Query for the contract with the given contract key.
+--
+-- Returns `None` if there is no active party that the party is a stakeholder on.
```
Comment (hurryabit) [inline daml-script/daml/Daml/Script.daml:132]:
```suggestion
-- Returns `None` if there is no active contract that the party is a stakeholder on.
```

**Review by hurryabit — APPROVED** [review]
Very nice. Thank you.

@cocreature We could back this by a more efficient implementation when going through the JSON API. Is that something worth doing (yet)?

**Review by cocreature — CHANGES_REQUESTED** [review]
Thanks for the PR, see https://github.com/digital-asset/daml/issues/7340 for my thoughts on this. I’d like to do this differently so we can handle it more efficiently in DAML Studio and in the JSON API (and it should be called `queryContractKey`)

**Comment by bame-da** [comment]
Duplicate of #7340

**Pull request closed without merging**